### PR TITLE
add optional 'forward' argument to lax.scan

### PR DESCRIPTION
**Pull request by mattjj**

### CODE REVIEW

**Comment by jekbradbury** [comment]
If we didn't have the param already, we'd probably call this `reverse`, right? Maybe we should switch the param too?

**Comment by mattjj** [comment]
I'm not sure. Is "reverse" better than "forward"? They seem symmetric to me.

File: jax/lax/lax_control_flow.py [hunk 1]
```diff
@@ -921,13 +924,18 @@ def scan(f, init, xs, length=None):
   if jax.api._jit_is_disabled():
     carry = init
     ys = []
+    if not forward:
```
Comment (jekbradbury) [inline jax/lax/lax_control_flow.py:927]: Would a backwards for loop work? (and then reverse `y` in Python before passing it to `stack`?)

Comment (mattjj) [reply jax/lax/lax_control_flow.py:927]: Yes that would work, and perhaps make debugging easier. I'll give it a shot.

**Comment by jekbradbury** [comment]
I made the disable-jit edit, and switched to `reverse`. I hope that's okay? (If not I'm happy to revert that commit and we can merge as is--if anything, making the changes made me care less about them!).

**Comment by jekbradbury** [comment]
@googlebot I consent.

**Comment by mattjj** [comment]
Wooo awesome! Yes, certainly okay by me, and I am very thankful that you did that!

**Pull request merged**